ospfd/ospf_ext.h

```diff
@@ -0,0 +1,196 @@
+ * |                                   |  |  Opaque LSA;
+ * ~              Values ...           ~  |  Values might be structured as a set
+ * |                                   |  V  of sub-TLVs.
+ * +--------+--------+--------+--------+ ---
+ */
+
+/* Global use constant numbers */
+
+#define	MAX_LEGAL_EXT_INSTANCE_NUM	(0xffff)
+#define LEGAL_EXT_INSTANCE_RANGE(i)	(0 <= (i) && (i) <= 0xffff)
+
+/* Flags to manage Extended Link/Prefix Opaque LSA */
+#define EXT_LPFLG_LSA_INACTIVE          0x00
+#define EXT_LPFLG_LSA_ACTIVE            0x01
+#define EXT_LPFLG_LSA_ENGAGED           0x02
+#define EXT_LPFLG_LSA_LOOKUP_DONE       0x04
+#define EXT_LPFLG_LSA_FORCED_REFRESH    0x08
+#define EXT_LPFLG_FIB_ENTRY_SET         0x10
+
+/*
+ * Following section defines TLV (tag, length, value) structures,
+ * used in Extended Prefix/Link Opaque LSA.
+ */
+
+/* Extended Prefix TLV Route Types */
+#define EXT_TLV_PREF_ROUTE_UNSPEC	0
+#define EXT_TLV_PREF_ROUTE_INTRA_AREA	1
+#define EXT_TLV_PREF_ROUTE_INTER_AREA	3
+#define EXT_TLV_PREF_ROUTE_AS_EXT	5
+#define EXT_TLV_PREF_ROUTE_NSSA_EXT	7
+
+/* Extended Prefix and Extended Prefix Range TLVs'
+ * Address family flag for IPv4 */
+#define EXT_TLV_PREF_AF_IPV4		0
+
+/* Extended Prefix TLV Flags */
+#define EXT_TLV_PREF_AFLG		0x80
+#define EXT_TLV_PREF_NFLG		0x40
+
+/* Extended Prefix Range TLV Flags */
+#define EXT_TLV_PREF_RANGE_IAFLG	0x80
+
+/* ERO subtlvs Flags */
+#define EXT_SUBTLV_ERO_LFLG		0x80
+
+/* Extended Prefix TLV see RFC 7684 section 2.1 */
+#define EXT_TLV_PREFIX			1
+#define EXT_TLV_PREFIX_SIZE		8
+struct ext_tlv_prefix {
+	struct tlv_header header;
+	u_int8_t route_type;
+	u_int8_t pref_length;
+	u_int8_t af;
+	u_int8_t flags;
+	struct in_addr address;
+};
+
+/* Extended Link TLV see RFC 7684 section 3.1 */
+#define EXT_TLV_LINK			1
+#define EXT_TLV_LINK_SIZE		12
+struct ext_tlv_link {
+	struct tlv_header header;
+	u_int8_t link_type;
+	u_int8_t reserved[3];
+	struct in_addr link_id;
+	struct in_addr link_data;
+};
+
+/* Remote Interface Address Sub-TLV, Cisco experimental use Sub-TLV */
+#define EXT_SUBTLV_RMT_ITF_ADDR         32768
+#define EXT_SUBTLV_RMT_ITF_ADDR_SIZE	4
+struct ext_subtlv_rmt_itf_addr {
+	struct tlv_header header;
+	struct in_addr value;
+};
+
+/* Internal structure to manage Extended Link/Prefix Opaque LSA */
+struct ospf_ext_lp {
+	bool enabled;
+
+	/* Flags to manage this Extended Prefix/Link Opaque LSA */
+	u_int32_t flags;
+
+	/* Scope is area Opaque Type 10 or AS Opaque LSA Type 11 for
+	 * Extended Prefix and area Opaque Type 10 for Extended Link */
+	u_int8_t scope;
+
+	/* area pointer if flooding is Type 10 Null if flooding is AS scope */
+	struct ospf_area *area;
+	struct in_addr area_id;
+
+	/* List of interface with Segment Routing enable */
+	struct list *iflist;
+};
+
+/* Structure to aggregate interfaces information for Extended Prefix/Link */
+struct ext_itf {
+	/* 24-bit Opaque-ID field value according to RFC 7684 specification */
+	u_int32_t instance;
```

Comment:
I believe we need to use uint32_t, uint8_t, uint16_t in our code.

Reply:
Hum, `grep uint32_t ospfd/* | wc -l` in ospfd directory give only 11 while `grep u_int32_t ospfd/* | wc -l`give 310.
So, I follow current code. If I change for uint32_t and co. I think we must do that globally on all ospfd files.

Reply:
Please use C99 standard fixed width types moving forward. Mass conversion to the standard was brought up at one point (#1132) but shot down so we're doing this the slow way. These typedefs are nonstandard and no longer necessary so we are trying to phase them out.

Reply:
OK. I will change them on new files. But, when the PR #1132 you mention will be merge ? Do I need this old PR merged before changing ? or change now and this old PR will be aplpy later ?